### DISCUSSION TITLE
What am I supposed to do when PENDING on client side?

### DISCUSSION THREAD

**Discussion by gganjang** [microsoft/msquic#1201] [Q&A] [answered]
I'm digging into 'interop.cpp' example code to learn how to use.

I've run the binary with 'quicinterop.exe -target:msquic' options, but never succeeded. After some investigations, I figured out that MsQuic->ConnectionStart function in 'ConnectToServer' returns quic status, PENDING.

Now, what am I supposed to do to proceed? why does it give PENDING status?
+) I checked 'connection.c' and it was giving me QUIC_API_TYPE_CONN_SHUTDOWN after QUIC_API_TYPE_STRM_SEND

**Comment by nibanks** [comment, marked as answer]
Generally, the MsQuic API returns PENDING to indicate that the action isn't completely done yet. `ConnectionStart` just queues the work to the QUIC worker thread and then returns PENDING. From there, the worker will resolve DNS as necessary and then start the QUIC handshake. Eventually you would get a connected or shutdown event to know if the connection succeeded or not.

As far as debugging why things aren't working for you, I'd recommend two things:

1. Make sure to update to the latest. We're constantly fixing bugs (though this is unlikely your problem).
2. Find the reason for the connection shutdown. That should give you a clue as to what's happening.

**Reply by gganjang** [reply]
Thanks for the help. I will try to figure it out as soon as I can and update the progress

**Reply by gganjang** [reply]
Here's update on my work with MsQuic.
I gave up on the sample code but built my own project using MsQuic. In my project, connection status is PENDING until a stream is created by either host or client. As soon as the stream is created, everything starts to work fine.
